modules/apps/29-fee/ibc_middleware_test.go

```diff
@@ -607,166 +607,220 @@ func (suite *FeeTestSuite) TestOnRecvPacket() {
+			"success: channel is not fee not enabled",
 			func() {
 				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
 				ack = ibcmock.MockAcknowledgement.Acknowledgement()
-
-				expectedBalance = originalBalance
 			},
-			false,
 			true,
+			func() {},
 		},
 		{
 			"success: fee module is disabled, skip fee logic",
 			func() {
 				lockFeeModule(suite.chainA)
-				expLocked = true
-
-				expectedBalance = originalBalance
 			},
-			false,
 			true,
+			func() {
+				suite.Require().Equal(true, suite.chainA.GetSimApp().IBCFeeKeeper.IsLocked(suite.chainA.GetContext()))
+			},
 		},
 		{
-			"fail on distribute receive fee (blocked address)",
+			"success: fail to distribute recv fee (blocked address), returned to refund account",
 			func() {
 				blockedAddr := suite.chainA.GetSimApp().AccountKeeper.GetModuleAccount(suite.chainA.GetContext(), transfertypes.ModuleName).GetAddress()
 
-				ack = types.IncentivizedAcknowledgement{
-					AppAcknowledgement:    ibcmock.MockAcknowledgement.Acknowledgement(),
-					ForwardRelayerAddress: blockedAddr.String(),
-				}.Acknowledgement()
+				// reassign ack.ForwardRelayerAddress to a blocked address
+				ack = types.NewIncentivizedAcknowledgement(blockedAddr.String(), ibcmock.MockAcknowledgement.Acknowledgement(), true).Acknowledgement()
+
+				// retrieve the relayer acc balance and add the expected ack fees
+				relayerAccBalance := sdk.NewCoins(suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), relayerAddr, sdk.DefaultBondDenom))
+				expPayeeAccBalance = relayerAccBalance.Add(packetFee.Fee.AckFee...)
 
-				expectedRelayerBalance = packetFee.Fee.AckFee
-				expectedBalance = expectedBalance.Add(packetFee.Fee.RecvFee...)
+				// retrieve the refund acc balance and add the expected recv fees and timeout fees
+				refundAccBalance := sdk.NewCoins(suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAddr, sdk.DefaultBondDenom))
+				expRefundAccBalance = refundAccBalance.Add(packetFee.Fee.RecvFee...).Add(packetFee.Fee.TimeoutFee...)
 			},
-			false,
 			true,
+			func() {
+				// assert that the packet fees have been distributed
+				found := suite.chainA.GetSimApp().IBCFeeKeeper.HasFeesInEscrow(suite.chainA.GetContext(), packetID)
+				suite.Require().False(found)
+
+				relayerAccBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), relayerAddr, sdk.DefaultBondDenom)
+				suite.Require().Equal(expPayeeAccBalance, sdk.NewCoins(relayerAccBalance))
+
+				refundAccBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAddr, sdk.DefaultBondDenom)
+				suite.Require().Equal(expRefundAccBalance, sdk.NewCoins(refundAccBalance))
+			},
 		},
 		{
-			"fail on no distribution by escrow account out of balance",
+			"fee distribution fails by escrow account out of balance",
```

Comment:
nit: just keeping in line with semantics from above 
```suggestion
			"fail: fee distribution fails by escrow account out of balance",
```